tripy/tripy/frontend/utils.py

```diff
@@ -212,9 +240,28 @@ def impl(func):
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
             from tripy.frontend.tensor import Tensor
+            from tripy.frontend.shape import Shape, ShapeScalar
 
             all_args = utils.merge_function_arguments(func, *args, **kwargs)
 
+            # TODO (#233): Disallow mixing Tensor/Shape.
```

Comment:
Per previous discussion, I think this policy is reasonable as a stopgap measure, especially since we could simply not use the decorator if we run into a case where we think we should be able to mix types. I think the comment should mention the bit about ShapeScalars likely being broadcast into shapes, as the max type policy wouldn't work otherwise.

Reply:
Updated the comment.